test/connectors/MarriageAllowanceDESConnectorTest.scala

```diff
@@ -0,0 +1,434 @@
+          val ninoWithSpaces = Nino(generatedNino.formatted)
+
+          val result = await(connector.findRecipient(request))
+
+          result shouldBe Right(expectedResult)
+        }
+      }
+    }
+
+    "contain the correct headers to send to DES" in new FindRecipientSetup {
+
+      val uuidRegex = """[0-9a-fA-F]{8}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{12}"""
+
+      val reasonCode = ProcessingOK
+      val returnCode = ProcessingOK
+
+      val json = expectedJson(reasonCode, returnCode)
+
+
+      server.stubFor(
+        post(urlEqualTo(url))
+          .willReturn(ok(json.toString()))
+      )
+
+      val result = await(connector.findRecipient(request))
+
+      server.verify(postRequestedFor(urlEqualTo(url))
+        .withHeader("Authorization", equalTo(connector.urlHeaderAuthorization))
+        .withHeader("Environment", equalTo(connector.urlHeaderEnvironment))
+        .withHeader("CorrelationId", matching(uuidRegex))
+      )
+    }
+
+
+    "return a ResponseValidator error given non valid Json" in { new FindRecipientSetup {
+
+      val nonValidJson = s"""{
+          "Jfwk1012FindCheckPerNoninocallResponse": {
+            "Jfwk1012FindCheckPerNoninoExport": {
+              "@exitStateType": "0",
+              "@exitState": "0",
+              "OutItpr1Person": {
+                "InstanceIdentifier": $instanceIdentifier
+              },
+              "OutWCbdParameters": {
+                "SeverityCode": "W",
+                "DataStoreStatus": "S",
+                "OriginServid": 9999,
+                "ContextString": "ITPR1311_PER_DETAILS_FIND_S",
+                "ReturnCode": 1,
+                "ReasonCode": 1,
+                "Checksum": "a234jnjbhr9ui83"
+              }
+            }
+          }
+      }"""
+
+      server.stubFor(
+        post(urlEqualTo(url))
+          .willReturn(ok(nonValidJson.toString()))
+      )
+
+      val result = await(connector.findRecipient(request))
+
+      result shouldBe Left(ResponseValidationError)
+
+      }
+    }
+
+    "return a Too Many Requests error type when a TOO_MANY_REQUESTS error is received " in { new FindRecipientSetup {
+        server.stubFor(
+          post(urlEqualTo(url))
+            .willReturn(aResponse().withStatus(TOO_MANY_REQUESTS))
+        )
+
+        val result = await(connector.findRecipient(request))
+
+        result shouldBe Left(TooManyRequestsError)
+
+       }
+    }
+
+
+    "return a BadRequest error type" when {
+
+      "a BadRequest is received " in { new FindRecipientSetup {
+          server.stubFor(
+            post(urlEqualTo(url))
+              .willReturn(aResponse().withStatus(BAD_REQUEST).withBody("Submission has not passed validation"))
+          )
+
+          val result = await(connector.findRecipient(request))
+
+          result shouldBe Left(BadRequestError)
+        }
+      }
+    }
+
+
+    "return a TimeOutError " when {
```

Comment:
We test the exception and 499 response code, but not the 504 response code - which is different